### PR TITLE
Codeforces White Sheet Bronze

### PR DESCRIPTION
**Pull request by back2sqr1**
Added python, java, and cpp implementations as well as my explanation of the solution

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
make sure to edit metadata as well

File: solutions/cf-1216C.mdx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+				return st.nextToken();
+			} catch (Exception e) { }
+			return null;
+		}
+		public int nextInt() { return Integer.parseInt(next()); }
+		public double nextDouble() { return Double.parseDouble(next()); }
+		public long nextLong() { return Long.parseLong(next()); }
+	}
+  static class Rect{
+    int x1,y1,x2,y2;
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:189]:
```suggestion
    int x1, y1, x2, y2;
```
space variables so its easy to read

File: solutions/cf-1216C.mdx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      x1 = a;
+      y1 = b;
+      x2 = c;
+      y2 = d;
+    }
+  	int area()
+  	{
+  		return (x2-x1)*(y2-y1);
+  	}
+  }
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:200]: something about the brackets is off putting they aren't aligned with the code so fix it

File: solutions/cf-1216C.mdx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  		a.y2 = b.y1;
+  		a.y1 = Math.min(a.y1, b.y1); // if b covers A entirely
+  	}
+  	return a;
+  }
+}
+```
+
+</JavaSection>
+
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:234]:
```suggestion
```

File: solutions/cf-1216C.mdx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    Rect b = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    Rect c = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    a = cut(a, b);
+  	a = cut(a, c);
+  	if(a.x2-a.x1==0 || a.y2-a.y1==0) //area!=0
+  	System.out.println("NO");
+  	else
+  	System.out.println("YES");
+    io.close();
+  }
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:160]: wierd indentation

File: solutions/cf-1216C.mdx [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+public class Main {
+  
+  public static void main(String[] args) throws IOException {
+    Kattio io = new Kattio();
+		Rect a = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    Rect b = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    Rect c = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:152]:
```suggestion
    Kattio io = new Kattio();
    Rect a = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
    Rect b = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
    Rect c = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
```
fix rect a indentation

File: solutions/cf-1216C.mdx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    a = cut(a, b);
+  	a = cut(a, c);
+  	if(a.x2-a.x1==0 || a.y2-a.y1==0) //area!=0
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:155]:
```suggestion
  	if(a.x2-a.x1==0 || a.y2-a.y1==0) //area!=0
```
```suggestion
  	if(a.x2 - a.x1 == 0 || a.y2 - a.y1 == 0) //area!=0
```
spacing

File: solutions/cf-1216C.mdx [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+  	System.out.println("NO");
+  	else
+  	System.out.println("YES");
```
Comment (sathu-beep) [inline solutions/cf-1216C.mdx:158]:
```suggestion
  	System.out.println("NO");
  	else
  	System.out.println("YES");
```
add brackets and fix indentation of the else

**Review by sathu-beep — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
yo what

**Pull request closed without merging**